src/client/app/services/services.js

```diff
@@ -0,0 +1,65 @@
+interview_coding_app.factory('FavStuffService', ['$http', '$q', '$httpParamSerializerJQLike', function ($http, $q, $httpParamSerializerJQLike) {
+        var FavStuffServiceFactory = {};
+        var _getAllFavStuff = function () {
+            var deferred = $q.defer();
```

Comment:
There is another way to handle promises in AngularJS. 
http://www.codelord.net/2015/09/24/%24q-dot-defer-youre-doing-it-wrong/

(Sorry for the _mostly_ offending title, but interesting read nonetheless)